.github/CONTRIBUTING.md

```diff
@@ -1,34 +1,85 @@
-a Contributor License Agreement (CLA). Please see our [CLA repo](http://github.com/IBM-Swift/CLA) for more information.
 
-This clarifies the intellectual property license granted with any contribution. It is for your protection as a
-Contributor as well as the protection of IBM and its customers; it does not
-change your rights to use your own Contributions for any other purpose.
+2. Make a change, usually by tackling an issue. 
 
-### Coding standards
+**Tip:** If you are on macOS, Xcode can be a great way to work on Kitura. You'll need to open a Terminal window, and then navigate to the root of the Kitura git project. Then, type the following command into Terminal:
 
-Please ensure you follow [the Kitura coding standards](https://github.com/IBM-Swift/Kitura/blob/master/Documentation/CodeConventions.md)
+   `$ swift package generate-xcodeproj`
 
-Please note:
+This will generate an Xcode Project, simply open the .xcodeproj file with:
 
- - all files must have the Apache license in the header.
- - all PRs must have passing builds for all operating systems.
+   `$ open *.xcodeproj`
+
+Unfortunately, Xcode is macOS only, and Linux will need to use a different text editor or IDE.
+
+3. Ensure all tests pass with your changes by running `swift test`. If there is any new funtionality introduced by your change, new test case(s) may be required. If you need any help writing test cases, contact us on slack.
+
+4. If they all pass, open a Pull Request following the guidelines below.
+
+## Pull Requests
+
+Issues that would be a good place to start for someone new to GitHub will have the label "Good First Issue". If there aren't any tagged issues, post in Slack and the team would be happy to help you get started.
+
+ When opening a new PR, please follow the [PR template](PULL_REQUEST_TEMPLATE.md).
+
+To start working on an issue, pick one from [here](https://github.com/IBM-Swift/Kitura/issues) and do the following:
+
+1. Fork the repository to your own account, and create a branch for the issue you're working on. A good branch name could be as simple as issue.ISSUENUMBER or it can be more descriptive if you like.
+
+2. Work on the issue. This part is harder to write a guide for but some general tips are:
+
+   * Please respect the [original style guide](https://github.com/github/swift-style-guide).
+   * Create minimal differences - try not to reformat the code. Open a separate PR if you feel the source codes structure needs changing.
+   * Check for unneccessary white space using `git diff --check` before you commit.
+
+3. Verify all the tests still pass with your changes by running `swift test`. If your code introduces a new test case, please add it and ensure it passes too.
+
+4. Push your fork and then open a PR against the original repos **master** branch.
+
+5. Keep an eye on your inbox - we may request clarification or ask for changes if there's a problem.
+
+## Coding Style
```

Comment:
Let's include this line from before: `-Please ensure you follow [the Kitura coding standards](https://github.com/IBM-Swift/Kitura/blob/master/Documentation/CodeConventions.md)`, and explicitly mention `swift lint`.